Wikipedia/Code/ArticleViewController+LinkPreviewing.swift

```diff
@@ -146,3 +146,43 @@ extension ArticleViewController {
         commitPreview(of: previewingViewController)
     }
 }
+
+// MARK: Peek/Pop for Lead Image of ArticleVC (iOS 13 and later - no equivalent functionality on iOS 12 and prior)
+@available(iOS 13.0, *)
+extension ArticleViewController: UIContextMenuInteractionDelegate {
+    func contextMenuInteraction(_ interaction: UIContextMenuInteraction, configurationForMenuAtLocation location: CGPoint) -> UIContextMenuConfiguration? {
+        // If gallery has not been opened on that article, self.mediaList is nil - and we need to fake the media list
+        guard let mediaList = self.mediaList ?? oneImageMediaList(from: leadImageView.wmf_imageURLToFetch) else {
+            return nil
+        }
+        let previewProvider: UIContextMenuContentPreviewProvider = {
+            let galleryVC = self.getGalleryViewController(for: nil, in: mediaList)
+            galleryVC.setOverlayViewTopBarHidden(true)
+            return galleryVC
+        }
+        return UIContextMenuConfiguration(identifier: nil, previewProvider: previewProvider, actionProvider: nil)
+    }
+
+    func contextMenuInteraction(_ interaction: UIContextMenuInteraction, willPerformPreviewActionForMenuWith configuration: UIContextMenuConfiguration, animator: UIContextMenuInteractionCommitAnimating) {
+        animator.addCompletion {
+            self.showLeadImage()
+        }
+    }
+
+    private func oneImageMediaList(from url: URL?) -> MediaList? {
```

Comment:
Would it make sense to have this as an initializer on `MediaList`?

Reply:
Good call, done. Tried to leave a clear comment on the new initializer regarding what it is used for.